sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2015,6 +2052,8 @@ pi_result piQueueRelease(pi_queue Queue) {
 
     zePrint("piQueueRelease NumTimesClosedFull %d, NumTimesClosedEarly %d\n",
             Queue->NumTimesClosedFull, Queue->NumTimesClosedEarly);
+    Queue->PiQueueMutex.unlock();
```

Comment:
remove this line?!
actually you continue to operate on the queue that is not locked, which isn't right.
I think you have to keep the lock until you are about to delete it.

Reply:
Oops. My mistake.
Fixed it. Thanks.